packages/agreement-email-sender/src/index.ts

```diff
@@ -40,16 +40,25 @@ export async function processMessage({
   });
   loggerInstance.debug(decodedMessage);
 
-  match(decodedMessage).with(
-    { event_version: 2, type: "AgreementActivated" },
-    async ({ data: { agreement } }) => {
-      if (agreement) {
-        await sendAgreementEmail(agreement, readModelService, emailManager);
-      } else {
-        throw missingKafkaMessageDataError("agreement", decodedMessage.type);
+  await match(decodedMessage)
+    .with(
+      { event_version: 2, type: "AgreementActivated" },
```

Comment:
I suggest an exhaustive match (like [here](https://github.com/pagopa/interop-be-monorepo/blob/main/packages/authorization-updater/src/index.ts#L205)) to help the introduction of future events